client/js/render.js

```diff
@@ -39,9 +39,11 @@ function buildChatMessage(data) {
 	const chan = chat.find(target);
 	let template = "msg";
 
-	if (!data.msg.highlight && !data.msg.self && (type === "message" || type === "notice") && options.highlights.some(function(h) {
-		return data.msg.text.toLocaleLowerCase().indexOf(h.toLocaleLowerCase()) > -1;
-	})) {
+	// See if any of the custom highlight regexes match
+	if (!data.msg.highlight && !data.msg.self
+		&& (type === "message" || type === "notice")
+		&& options.highlightsRE
```

Comment:
Are string checks more expensive then checking if something is true? Or do you have some other reason?